lib/tasks/puppet.rake

```diff
@@ -57,4 +57,32 @@ namespace :puppet do
       end
     end
   end
+
+  namespace :migrate do
+    desc "
+    Migrate Puppet provisioning parameters from ENC to Host parameters.
+    This task is relevant if you want to continue using Puppet provisioning,
+    but do not want to install the Puppet plugin and get the full ENC funcionality."
+
+    task :to_parameters => :environment do
+      class FakeEnv < ::ApplicationRecord
+        self.table_name = 'environments'
+      end
+
+      User.current = User.anonymous_console_admin
+      envs = Hash[FakeEnv.all.pluck(:id, :name)]
+
+      Hostgroup.where.not(environment_id: nil).find_each(batch_size: 100) do |hg|
+        GroupParameter.create(hostgroup: hg, name: 'puppet_environment', value: envs[hg.environment_id])
+        GroupParameter.create(hostgroup: hg, name: 'puppet_server', value: hg.puppet_server)
+        GroupParameter.create(hostgroup: hg, name: 'puppet_ca_server', value: hg.puppet_ca_server)
+      end
+
+      Host.where.not(environment_id: nil).find_each(batch_size: 100) do |host|
```

Comment:
What about hosts that have puppet server/ca set but inherit environment from their hg?

Reply:
inheriting environment does set it on the Host on `save`, doesn't it?

Reply:
no, it is not, environment needs to be set to set proxy and proxy needs to be set when env set, when env is cleared, proxies get cleared. You just can not to have CA :)